### PR TITLE
enh(dashboards): correct ACLs on delete contact/contactgroup shares

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to correctly check ACLs while deleting a contact/contactgroup shares on a dashboard.

This also remove many no more used endpoints

**Fixes** # MON-47054

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

## Summary by CodeRabbit

- **Documentation**
  - Removed API documentation references and update functionalities for dashboard contact and contact group access rights.

- **New Features**
  - Added error handling for unauthorized access based on user groups.

- **Bug Fixes**
  - Enhanced security checks to verify if a contact or contact group exists in user's access groups before allowing modifications to dashboard shares.

### CODE REVIEW

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -18916,3 +18916,9 @@ msgstr "Standard"
 
 msgid "Condensed"
 msgstr "Condensé"
+
+msgid "The contact group [%s] is not in your access groups"
+msgstr "Le groupe de contact [%s] ne fait pas partie de vos groupes d'accès"
+
+msgid "The user [%s] is not in your access groups"
+msgstr "L'utilisateur [%s] ne fait pas partie de vos groupes d'accès"
```
Comment (a-launois) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18924]: I'm not a fan of the wording, we usualy just say that the user/usergroup doesn't exist or was not found without specifying that its due to the user rights.

**Review by a-launois — CHANGES_REQUESTED** [review]

File: centreon/src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php [hunk 2]
```diff
@@ -74,6 +74,37 @@ public function exists(int $contactGroupId): bool
...
+        $accessGroupIdsAsString = implode(',', array_keys($bind));
+
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<SQL
+                SELECT 1
+                FROM `:db`.contactgroup cg
+                     INNER JOIN `:db`.acl_group_contactgroups_relations gcgr
+                               ON cg.cg_id = gcgr.cg_cg_id
+                WHERE cg.cg_id = :contactGroupId
+                    AND gcgr.acl_group_id IN ({$accessGroupIdsAsString}
```
Comment (a-launois) [inline centreon/src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php:96]:
```suggestion
                    AND gcgr.acl_group_id IN ({$accessGroupIdsAsString})
```

Comment (a-launois) [reply centreon/src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php:96]: Also you can use the SqlMultipleBindTrait for that

**Review by a-launois — APPROVED** [review]

File: centreon/src/Core/Dashboard/Application/UseCase/DeleteContactDashboardShare/DeleteContactDashboardShare.php [hunk 3]
```diff
@@ -145,6 +150,13 @@ private function deleteContactShareAsContact(Dashboard $dashboard, int $contactI
             );
         }
 
+        $accessGroups = $this->readAccessGroupRepository->findByContact($this->contact);
+        $accessGroupIds = array_map(static fn (AccessGroup $accessGroup): int => $accessGroup->getId(), $accessGroups);
```
Comment (adr-mo) [inline centreon/src/Core/Dashboard/Application/UseCase/DeleteContactDashboardShare/DeleteContactDashboardShare.php:154]: Could be done in one line as the `$accessGroups` variable is not used elsewhere (that is not mandatory)

File: centreon/src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php [hunk 4]
```diff
@@ -74,6 +74,37 @@ public function exists(int $contactGroupId): bool
         return (bool) $statement->fetchColumn();
     }
 
+    public function existsInAccessGroups(int $contactGroupId, array $accessGroupIds): bool
```
Comment (adr-mo) [inline centreon/src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php:77]:
```suggestion
    /**
     * @inheritDoc
     */
    public function existsInAccessGroups(int $contactGroupId, array $accessGroupIds): bool
```

File: centreon/src/Core/Dashboard/Application/UseCase/DeleteContactGroupDashboardShare/DeleteContactGroupDashboardShare.php [hunk 5]
```diff
@@ -151,6 +154,13 @@ private function deleteContactGroupShareAsContact(Dashboard $dashboard, int $con
             );
         }
 
+        $accessGroups = $this->readAccessGroupRepository->findByContact($this->contact);
+        $accessGroupIds = array_map(static fn (AccessGroup $accessGroup): int => $accessGroup->getId(), $accessGroups);
```
Comment (adr-mo) [inline centreon/src/Core/Dashboard/Application/UseCase/DeleteContactGroupDashboardShare/DeleteContactGroupDashboardShare.php:158]: Same comment as before

**Review by adr-mo — APPROVED** [review]

File: centreon/src/Core/Dashboard/Application/UseCase/DeleteContactGroupDashboardShare/DeleteContactGroupDashboardShare.php (hunk 5 above)
Comment (jeremyjaouen) [reply centreon/src/Core/Dashboard/Application/UseCase/DeleteContactGroupDashboardShare/DeleteContactGroupDashboardShare.php:158]: Seen with adr-mo

File: centreon/src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php (hunk 4 above)
Comment (jeremyjaouen) [reply centreon/src/Core/Contact/Infrastructure/Repository/DbReadContactGroupRepository.php:77]: Seen with adr-mo

File: centreon/src/Core/Dashboard/Application/UseCase/DeleteContactDashboardShare/DeleteContactDashboardShare.php (hunk 3 above)
Comment (jeremyjaouen) [reply centreon/src/Core/Dashboard/Application/UseCase/DeleteContactDashboardShare/DeleteContactDashboardShare.php:154]: Seen with adr-mo

**Pull request merged**